internal/resource/service.go

```diff
@@ -109,19 +109,77 @@ func applySvcOverride(svc *corev1.Service, override *rabbitmqv1beta1.Service) er
 	return nil
 }
 
-func (builder *ServiceBuilder) updatePorts(servicePorts []corev1.ServicePort) []corev1.ServicePort {
+func (builder *ServiceBuilder) generateServicePortsMapOnlyTLSListeners() map[string]corev1.ServicePort {
+	servicePortsMap := map[string]corev1.ServicePort{
+		"amqps": {
+			Protocol:   corev1.ProtocolTCP,
+			Port:       5671,
+			TargetPort: intstr.FromInt(5671),
+			Name:       "amqps",
+		},
+		"management-tls": {
+			Protocol:   corev1.ProtocolTCP,
+			Port:       15671,
+			TargetPort: intstr.FromInt(15671),
+			Name:       "management-tls",
+		},
+	}
+
+	if builder.Instance.AdditionalPluginEnabled("rabbitmq_stomp") {
+		servicePortsMap["stomps"] = corev1.ServicePort{
+			Protocol:   corev1.ProtocolTCP,
+			Port:       61614,
+			Name:       "stomps",
+			TargetPort: intstr.FromInt(61614),
+		}
+	}
+	if builder.Instance.AdditionalPluginEnabled("rabbitmq_mqtt") {
+		servicePortsMap["mqtts"] = corev1.ServicePort{
+			Protocol:   corev1.ProtocolTCP,
+			Port:       8883,
+			Name:       "mqtts",
+			TargetPort: intstr.FromInt(8883),
+		}
+	}
+
+	if builder.Instance.MutualTLSEnabled() {
+		if builder.Instance.AdditionalPluginEnabled("rabbitmq_web_stomp") {
+			servicePortsMap["stomps"] = corev1.ServicePort{
+				Protocol:   corev1.ProtocolTCP,
+				Port:       15673,
+				Name:       "web-stomp-tls",
+				TargetPort: intstr.FromInt(15673),
+			}
+		}
+		if builder.Instance.AdditionalPluginEnabled("rabbitmq_web_mqtt") {
+			servicePortsMap["mqtts"] = corev1.ServicePort{
```

Comment:
Doesn't this override the MQTT TLS port defined a few lines above?

Reply:
Updated, thank you for catching it!